libs/aws/langchain_aws/chat_models/sagemaker_endpoint.py

```diff
@@ -0,0 +1,130 @@
+)
+
+from langchain_core.callbacks import CallbackManagerForLLMRun
+from langchain_core.language_models import (
+    BaseChatModel,
+)
+from langchain_core.messages import (
+    AIMessage,
+    BaseMessage,
+    HumanMessage,
+    SystemMessage,
+    merge_message_runs,
+)
+from langchain_core.outputs import ChatGeneration, ChatResult
+from pydantic import ConfigDict
+
+from langchain_aws.llms.sagemaker_endpoint import SagemakerEndpoint
+
+logger = logging.getLogger(__name__)
+
+
+class ChatSagemakerEndpoint(BaseChatModel, SagemakerEndpoint):
+    """A chat model that uses a HugguingFace TGI compatible SageMaker Endpoint."""
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of chat model."""
+        return "amazon_sagemaker_chat"
+
+    @classmethod
+    def is_lc_serializable(cls) -> bool:
+        """Return whether this model can be serialized by Langchain."""
+        return True
+
+    @classmethod
+    def get_lc_namespace(cls) -> List[str]:
+        """Get the namespace of the langchain object."""
+        return ["langchain", "chat_models", "sagemaker"]
+
+    @property
+    def lc_attributes(self) -> Dict[str, Any]:
+        attributes: Dict[str, Any] = {}
+
+        if self.region_name:
+            attributes["region_name"] = self.region_name
+
+        return attributes
+
+    model_config = ConfigDict(
+        extra="forbid",
+        populate_by_name=True,
+    )
+
+    def _generate(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> ChatResult:
+        _model_kwargs = self.model_kwargs or {}
+        _model_kwargs = {**_model_kwargs, **kwargs}
+        _endpoint_kwargs = self.endpoint_kwargs or {}
+        sagemaker_messages = _messages_to_sagemaker(messages)
+        logger.debug(f"input message to sagemaker: {sagemaker_messages}")
+        invocation_params = {
+            "EndpointName": self.endpoint_name,
+            "Body": self.content_handler.transform_input(
+                sagemaker_messages, _model_kwargs
+            ),
+            "ContentType": self.content_handler.content_type,
+            "Accept": self.content_handler.accepts,
+            **_endpoint_kwargs,
+        }
+
+        # If inference_compoent_name is specified, append it to invocation_params
+        if self.inference_component_name:
+            invocation_params["InferenceComponentName"] = self.inference_component_name
+
+        try:
+            response = self.client.invoke_endpoint(**invocation_params)
+        except Exception as e:
+            logging.error(f"Error raised by inference endpoint: {e}")
+            if run_manager is not None:
+                run_manager.on_llm_error(e)
+            raise e
+        logger.info(f"The message received from SageMaker: {response['Body']}")
+
+        response_message = self.content_handler.transform_output(response["Body"])
```

Comment:
Should we rather support BaseMessage/List<BaseMessage> out from the output handlers? Similar format for input handler? Trying to evaluate if we bring it closer to what ChatModels support in terms of input/outputs?